src/Engines/BaseEngine.cs

```diff
@@ -456,54 +460,36 @@ public virtual void OnShutdownRequest(Message message)
 
         #region Command Parsing
 
-        public virtual bool IsMagic(string input, out ISymbol symbol)
+        /// <summary>
+        ///      Returns <c>true</c> if a given input is a magic symbol.
+        ///      If this method returns true, then <c>symbol</c> will
+        ///      be populated with the resolution of the magic symbol,
+        ///      <c>commandInput</c> will be populated with the magic
+        ///      symbol and its arguments, and <c>remainingInput</c> will
+        ///      be populated with any subsequent commands in the cell.
+        /// </summary>
```

Comment:
This API documentation comment is quite helpful, but I think It may be helpful to provide a brief example here as well, since there's a fair bit going on.